test/specs/elements/Header/Header-test.js

```diff
@@ -0,0 +1,36 @@
+import React from 'react'
+import faker from 'faker'
+import Header from 'src/elements/Header/Header'
+import Icon from 'src/elements/Icon/Icon'
+import Image from 'src/elements/Image/Image'
+import H1 from 'src/elements/Header/HeaderH1'
+import H2 from 'src/elements/Header/HeaderH2'
+import H3 from 'src/elements/Header/HeaderH3'
+import H4 from 'src/elements/Header/HeaderH4'
+import H5 from 'src/elements/Header/HeaderH5'
+import H6 from 'src/elements/Header/HeaderH6'
+import Subheader from 'src/elements/Header/HeaderSubheader'
+import * as common from 'test/specs/commonTests'
+
+describe('Header', () => {
+  common.isConformant(Header)
+  common.hasSubComponents(Header, [
+    H1, H2, H3, H4, H5, H6, Subheader,
+  ])
+
+  describe('with icon', () => {
+    const icon = <Icon className='settings' />
+    const wrapper = mount(<Header className='icon' icon={icon} />)
```

Comment:
Had problems using shallow with the subcomponent approach:

Header.H1 doesn't render the _Header classNames it's self, so the tests 
were failing!

Levi Thomason wrote:
>
> In test/specs/elements/Header/Header-test.js
> <https://github.com/TechnologyAdvice/stardust/pull/278#discussion_r68427616>:
>
> > +import H3 from'src/elements/Header/HeaderH3'
> > +import H4 from'src/elements/Header/HeaderH4'
> > +import H5 from'src/elements/Header/HeaderH5'
> > +import H6 from'src/elements/Header/HeaderH6'
> > +import Subheader from'src/elements/Header/HeaderSubheader'
> > +import * as common from'test/specs/commonTests'
> > +
> > +describe('Header', () => {
> > + common.isConformant(Header)
> > + common.hasSubComponents(Header, [
> > + H1, H2, H3, H4, H5, H6, Subheader,
> > + ])
> > +
> > + describe('with icon', () => {
> > + const icon =<Icon className='settings' />
> > + const wrapper = mount(<Header className='icon' icon={icon} />)
>
> Prefer |shallow| rendering when possible to help isolate tests. I
> believe it should work here.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/TechnologyAdvice/stardust/pull/278/files/4317290b6fd2e4980cfd74e736287438f69d4c6c#r68427616>, 
>
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAWHszdyrD23bRL7cApTPalJO6Ls1oE9ks5qPAymgaJpZM4I9saB>.